pkg/auth/arn_test.go

```diff
@@ -73,6 +73,51 @@ func TestParseARN(t *testing.T) {
 	}
 }
 
+func TestParseResources(t *testing.T) {
+	cases := []struct {
```

Comment:
Please add cases for failure, bad inputs expecting certain errors (empty resource, not json format, not array, w/e)